### DISCUSSION TITLE
difficult error when fetching remote refs

### DISCUSSION THREAD

**Discussion by mjgaughan** [gitpython-developers/GitPython#1983] [Q&A]
Hello! 

I am running into an error when I try to fetch some refs from the remote of my repository. I have spent some time in the GitPython docs but am still unable to work my head around where I could be going wrong, thus I am opening the ticket. 

I'm running into issues with the line below, where repo is a git.Repo instance.
``
repo.remote().fetch('refs/pull/*/merge:refs/pull/*/merge')
`` 
I have pasted the partial traceback below: 

```
 File "collating_prs.py", line 12, in get_prs
    results = repo.remote().fetch('refs/pull/*/merge:refs/pull/*/merge')
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "lib/python3.11/site-packages/git/remote.py", line 1069, in fetch
    res = self._get_fetch_info_from_stderr(proc, progress, kill_after_timeout=kill_after_timeout)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/lib/python3.11/site-packages/git/remote.py", line 895, in _get_fetch_info_from_stderr
    proc.wait(stderr=stderr_text)
  File "/lib/python3.11/site-packages/git/cmd.py", line 834, in wait
    raise GitCommandError(remove_password_if_present(self.args), status, errstr)
git.exc.GitCommandError: Cmd('git') failed due to: exit code(1)
  cmdline: git fetch -v -- origin refs/pull/*/merge:refs/pull/*/merge
```
In a bash terminal, the command ``git fetch -v -- origin refs/pull/*/merge:refs/pull/*/merge`` works and returns the refs that I am trying to fetch. 

I'm not quire sure where to start looking at this and would appreciate any guidance. Thank you for your help!

**Comment by Byron** [comment]
I recommend to set [`GIT_PYTHON_TRACE=full`](https://gitpython.readthedocs.io/en/stable/tutorial.html#git-command-debugging-and-customization) to hopefully see more output.

The usual issue would be an authentication failure.

I hope that helps.

**Reply by mjgaughan** [reply]
Yes! Thank you --- it seems to be an authentication failure.
